### PR TITLE
Login/Logout System Integration

### PR DESCRIPTION
**Pull request by Lemayzeur**
- Logout by using token in header.

- Token sent in: Login/Register

Profile instances are useful. Implementation of the profile creation signal.

### CODE REVIEW

**Comment by madsenwjservius** [comment]
I prefer you to create a branch and push it to the main repository then make the PR from this branch to the main branch. This way it  is more easily for me to grab your changes locally

**Comment by Lemayzeur** [comment]
> I prefer you to create a branch and push it to the main repository then make the PR from this branch to the main branch. This way it is more easily for me to grab your changes locally

Since I already pushed on the main branch and your changes are on your local machine, just move them to a new branch, and grab my commits just to review the pull request... and for further changes, we will work on different branches and merge them at some points.


Dispatch roles and functionalities to avoid conflicts.

**Comment by thecassion** [comment]
@Chamane What will you do with this pull request . We want more functionalities

File: mlhaiti/accounts/models.py [hunk 1]
```diff
@@ -83,3 +87,9 @@ class Meta:
 	photo75 = models.ImageField(upload_to=object_directory_path('75x75'), blank=True)
 	photo37 = models.ImageField(upload_to=object_directory_path('37x37'), blank=True)
 	photo18 = models.ImageField(upload_to=object_directory_path('18x18'), blank=True)
+
+@receiver(post_save, sender=User)
+def create_auth_token(sender, instance=None, created=False, **kwargs):
+	if created:
+		Token.objects.create(user=instance)
+		Profile.objects.create(user=instance)
```
Comment (madsenwjservius) [inline mlhaiti/accounts/models.py:95]: Genyen yon bug nan jan chan "code" ki nan profile la jenere a. Mwen poko gadel an detay kisa ki koz li. Men loske mwen komante liy lan pa genyen pwoblem lan anko... Donk se li ki leve eksepsyon an

File: mlhaiti/accounts/api/views.py [hunk 2]
```diff
@@ -18,13 +20,15 @@ class UserCreateView(generics.CreateAPIView):
     def create(self, request, *args, **kwargs):
         serializer = self.serializer_class(data=request.data)
         if serializer.is_valid():
-            serializer.save()
+            user = serializer.save()
+            data = serializer.data 
+            data['token'] = user.auth_token.key
```
Comment (madsenwjservius) [inline mlhaiti/accounts/api/views.py:25]: Eske ou mete token sa pou tou authenticate user la loske li finn enskri. Mwen panse fok ou valide account lan avan. valide email avan sa selon avim. Mwen gendwa wrong tou. Eklerem plis

File: mlhaiti/accounts/api/views.py [hunk 3]
```diff
@@ -33,9 +37,86 @@ def create(self, request, *args, **kwargs):
...
+        logout(request) # Delete session
+        return Response(
+            {
+                'error':False,
+                "message": _("Successfully logged out."),
+                "data":{},
+                "status": 200
+            },
+            status=status.HTTP_200_OK
+        )
```
Comment (madsenwjservius) [inline mlhaiti/accounts/api/views.py:121]: Li pa ranvoye mesaj erè ou mandel pou ranvoye a loske ou pa pase token nan nan header a. li jis afiche saa:

{
      detail: "Authentication credentials were not provided."
}

File: mlhaiti/accounts/models.py (hunk 1 above)
Comment (Chamane) [reply mlhaiti/accounts/models.py:95]: Loske mwen pase token lan tou li stil bay menm erè a. Eske erè sa kote pam li ye ? eske se mw ki mal itilize postman pou logout user. Kitem konnen

File: mlhaiti/accounts/api/views.py (hunk 3 above)
Comment (madsenwjservius) [reply mlhaiti/accounts/api/views.py:121]: Loske mwen vle logout user a mwen rankontre menm pwoblem lan. Menm mesaj la

Comment (madsenwjservius) [reply mlhaiti/accounts/api/views.py:121]: Login nan mache byen. 👍🏾

Comment (Lemayzeur) [reply mlhaiti/accounts/api/views.py:121]: Yes, paske li se yon endpoint proteje.  Ou dwe pase token nan, nan header, si n pap fe logout ak token, it's useless to have a logout endpoint

File: mlhaiti/accounts/api/views.py (hunk 2 above)
Comment (Lemayzeur) [reply mlhaiti/accounts/api/views.py:25]: Depi user a fin enskri, fok ou tounen token nan ba li, pou pwochen reket li yo, li pa bezwen login anko nan yon lot call

File: mlhaiti/accounts/models.py (hunk 1 above)
Comment (Lemayzeur) [reply mlhaiti/accounts/models.py:95]: The code is unique, si w te gen profile deja ki gen yon kod ki vid, al gade si pa gen lot pwofil ki pa gen chan sa vid tou, avan update pa m nan. Swa mete kod yo manyelman pou itilizate ki te deja kreye avan update la, oubyen efase yo, se pa foseman yon bug, se jis kontrent unique=True a

**Review by Chamane — APPROVED** [review]

**Pull request merged**